app/main.py

```diff
@@ -2,0 +1,27 @@
+class Animal:
+    alive: list = []
+
+    def __init__(self, name: str, health: int = 100,
+                 hidden: bool = False) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @classmethod
+    def bite(cls, herbivore: Herbivore) -> None:
+        if not isinstance(herbivore, cls) and not herbivore.hidden:
+            herbivore.health -= 50
+        if herbivore.health <= 0:
+            super().alive.remove(herbivore)
```

Comment:
The use of `super().alive.remove(herbivore)` is incorrect. You should use `Animal.alive.remove(herbivore)` to access the class variable `alive`.